### ISSUE TITLE
Mono Invoke bug?

### ISSUE THREAD

**Issue by jackindojogja** [cheat-engine/cheat-engine#2665]
Several methods can be invoked natively via the cheat engine. But some methods cause the game to crash. 

However, I can invoke all methods via "MonoBehaviour.Invoke" without crash.

Is this a cheatengine bug, or is there something wrong with my script?

![FB_IMG_1692319989164](https://github.com/cheat-engine/cheat-engine/assets/20923433/e3eb59c1-4eaa-4686-8098-942963a23a5d)

**Comment by cheat-engine** [comment]
Does the method work properly when called from MonoBehaviour.Invoke ?
Because if the call causes an error MonoBehaviour.Invoke might handle it internally while a raw call won't

**Comment by jackindojogja** [comment]
> Does the method work properly when called from MonoBehaviour.Invoke ? Because if the call causes an error MonoBehaviour.Invoke might handle it internally while a raw call won't

Yes, MonoBehaviour.Invoke work without crashing for ALL method, but execution time is slow, the game stuttering for a while, than directly use "mono_invoke_method".

I wonder, is it possible to invoke mono natively from Assembler without Lua? is there any example for that?

**Comment by jackindojogja** [comment]
And also, MonoBehaviour.Invoke doesn't supppot arguments/parameters, only void method.

**Comment by BlankUserLing** [comment]
You need to make sure that you have obtained the methodId correctly

**Comment by jackindojogja** [comment]
> You need to make sure that you have obtained the methodId correctly

any example?

I already check with print('methodId'), it's have value.

**Comment by BlankUserLing** [comment]
> 
I need you to tell me the program name

**Comment by jackindojogja** [comment]
> > 
> 
> I need you to tell me the program name

Bleach Brave Souls

How methodId can be incorrect, which I can Invoke "MonoBehaviour.Invoke" with same script?
